labview/labview_rabbitmq.c

```diff
@@ -260,15 +431,24 @@ int lv_amqp_consume_message(int64_t conn_intptr, int timeout_sec, LStrHandle out
 
 	//amqp_maybe_release_buffers(conn);
 
-	status = lv_report_amqp_error(amqp_consume_message(conn, &envelope, &tval, 0),"Consuming message", error_description);
-	if (status != 1) {
+	status = lv_report_amqp_error(amqp_consume_message(conn, &envelope, &tval, 0), "Consuming message", error_description);
+	if (status != 1)
+	{
 		return status;
 	}
 
-	if (envelope.message.body.len > 0) {
+	if (envelope.message.body.len > 0)
+	{
 		copyBufferToLStrHandle(envelope.message.body.bytes, envelope.message.body.len, output);
 	}
 
+	// Check message headers
+	amqp_table_t *headers = &envelope.message.properties.headers;
+	if (headers->num_entries > 0)
+	{
+		getConcatenatedMessageHeaders(headers, cheaders_key, cheaders_value);
```

Comment:
here is where when you receive a message you unpack headers to string arrays of keys and values that are then available to LabVIEW. correct?

LGTM

Reply:
true